modules/samples/authenticators/components/org.wso2.carbon.identity.sample.extension.auth.endpoint/src/main/webapp/hwk.jsp

```diff
@@ -0,0 +1,140 @@
+<%--
+  ~ Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+  ~
+  ~ WSO2 Inc. licenses this file to you under the Apache License,
+  ~ Version 2.0 (the "License"); you may not use this file except
+  ~ in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  --%>
+<%@ page language="java" session="true" %>
+<%@ page import="java.util.Random" %>
+<%@ page import="java.net.URLDecoder" %>
+<%@ page import="java.nio.charset.StandardCharsets" %>
+<%@ page import="org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils" %>
+
+<%
+
+    String reqNumberStr = request.getParameter("timeBasedId");
+    String callbackUrl = request.getParameter("callbackUrl");
+    String oldNumStr = request.getParameter("oldNum");
+
+    /* This is for demonstration purposes only.
+    You need to use a SecureRandom generator to avoid security breaches. */
+    Random randomGenerator = new Random();
+    int nextNum = randomGenerator.nextInt(10000);
+
+    if (reqNumberStr != null && oldNumStr != null) {
+        try {
+            int reqNumber = Integer.parseInt(reqNumberStr);
+            int oldNum = Integer.parseInt(oldNumStr);
+            if (oldNum == reqNumber) {
+                if (callbackUrl != null) {
+                    String cbURL = URLDecoder.decode(callbackUrl, StandardCharsets.UTF_8.name());
+                    cbURL = FrameworkUtils.appendQueryParamsStringToUrl(cbURL, "success=true");
+                    /* This is for demonstration purposes only.
+                    You need to properly encode the parameters to avoid security breaches. */
+                    response.sendRedirect(cbURL);
+                }
+            }
+        } catch (NumberFormatException e) {
+            //Ignore
```

Comment:
What if this exception occurs ? Nobody notices where it went wrong ? No redirection and no error message either ?